catalog/harvest/catalog-harvest-api/src/main/java/org/codice/ddf/catalog/harvest/Harvester.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.harvest;
+
+/**
+ * A {@code Harvester} is responsible for fetching remote resources and notifying any registered
+ * {@link Listener}s of create, update and delete events.
+ *
+ * <p>In the event that a {@code Harvester} goes down, it must remember the last state of the
+ * external server it was harvesting from. Upon a {@code Harvester} coming back up, it should only
+ * notify registered listener of new events.
+ */
+public interface Harvester {
```

Comment:
❓Should we assign a unique ID to each harvester and pass it down to the listeners? This might be useful down the road when we start supporting rules, i.e., allow users to write rules based on what harvester the resource came from. This could also be handled by the listeners if we prefer, which may be a better abstraction.

Reply:
Since we are going with a 1:1 harvester and listener I don't think this is necessary right now, but definitely needs to be done in the future.